### PR TITLE
add killPedTask function

### PR DESCRIPTION
**Pull request by Proxy-99**
`killPedTask(ped, string taskType, int taskNumber, [bool gracefully = true])`

issue related #3806

exmaple kill jump task
```lua
function playerPressedKey(button, press)
       local moveState = getPedMoveState( localPlayer )

       if(moveState=="jump") then
           killPedTask(localPlayer, "primary", 3, true)
       end
end
addEventHandler("onClientKey", root, playerPressedKey)
```

### CODE REVIEW

File: Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp [hunk 1]
```diff
@@ -2493,3 +2494,15 @@ bool CLuaPedDefs::SetPedExitVehicle(CClientPed* pPed)
 {
     return pPed->ExitVehicle();
 }
+
+bool CLuaPedDefs::killPedTask(CClientPed* ped, std::string taskType, uint8_t taskNumber, bool gracefully)
```
Comment (TracerDS) [inline Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp]: `std::uint8_t` + `noexcept`

**Review by TracerDS — CHANGES_REQUESTED** [review]

**Review by TracerDS — APPROVED** [review]

File: Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp [hunk 2]
```diff
@@ -2493,3 +2494,17 @@ bool CLuaPedDefs::SetPedExitVehicle(CClientPed* pPed)
...
+bool CLuaPedDefs::killPedTask(CClientPed* ped, std::string taskType, std::uint8_t taskNumber, std::optional<bool> gracefully) noexcept
+{
+    if (taskType == "primary")            // PRIMARY 
+    {
+        return ped->KillTask(taskNumber, gracefully.value_or(true));
+    }
+    else if (taskType == "secondary")     // SECONDARY 
+    {
+        return ped->KillTaskSecondary(taskNumber, gracefully.value_or(true));
+    }
```
Comment (FileEX) [inline Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp]: If you want you can remove braces here

**Review by FileEX — APPROVED** [review]

File: Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp [hunk 3]
```diff
@@ -2493,3 +2494,14 @@ bool CLuaPedDefs::SetPedExitVehicle(CClientPed* pPed)
 {
     return pPed->ExitVehicle();
 }
+
+bool CLuaPedDefs::killPedTask(CClientPed* ped, std::string taskType, std::uint8_t taskNumber, std::optional<bool> gracefully) noexcept
```
Comment (TheNormalnij) [inline Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp]: `string_view` ?

File: Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp [hunk 4]
```diff
@@ -2493,3 +2494,14 @@ bool CLuaPedDefs::SetPedExitVehicle(CClientPed* pPed)
 {
     return pPed->ExitVehicle();
 }
+
+bool CLuaPedDefs::killPedTask(CClientPed* ped, std::string_view taskType, std::uint8_t taskNumber, std::optional<bool> gracefully) noexcept
```
Comment (TheNormalnij) [inline Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp]: It looks like you need enum instead of string here

Comment (Proxy-99) [reply Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp]: but getPedTask ( ped theMed, string priority, int taskType ) uses string so should we have similar structure?

**Comment by TracerDS** [comment]
What would happen if you kill ped task while being in the air (complex jump tasks)?

**Comment by Proxy-99** [comment]
> What would happen if you kill ped task while being in the air (complex jump tasks)?

u will fall stand still the animation will stop but you have to call it in each frame like onClientRender

```lua

function displayMyTask ()
killPedTask(localPlayer, "primary", 1)
    local x,y = 100,200
    for k=0,4 do
        local a,b,c,d = getPedTask ( localPlayer, "primary", k )
        dxDrawText ( "Primary task #"..k.." is "..tostring(a).." -> "..tostring(b).." -> "..tostring(c).." -> "..tostring(d).." -> ", x, y )
        y = y + 15
    end
    y = y + 15
    for k=0,5 do
        local a,b,c,d = getPedTask ( localPlayer, "secondary", k )
        dxDrawText ( "Secondary task #"..k.." is "..tostring(a).." -> "..tostring(b).." -> "..tostring(c).." -> "..tostring(d).." -> ", x, y )    
        y = y + 15
    end
end    
addEventHandler ( "onClientRender", root, displayMyTask )
```

**Comment by TracerDS** [comment]
> > What would happen if you kill ped task while being in the air (complex jump tasks)?
> 
> u will fall stand still the animation will stop but you have to call it in each frame like onClientRender

Would you still continue the jump but without an animation or would the jump be cut off and instead of making the jump you will fall straight down?
Also, is it possible to also clear the task queue?

**Comment by Proxy-99** [comment]
> > > What would happen if you kill ped task while being in the air (complex jump tasks)?
> > 
> > 
> > u will fall stand still the animation will stop but you have to call it in each frame like onClientRender
> 
> Would you still continue the jump but without an animation or would the jump be cut off and instead of making the jump you will fall straight down? Also, is it possible to also clear the task queue?

I do not know about jumping while in air do you mean jump after an object then fall in air?
I think the  task queue is cleared once the task is killed or no task is active consider empty

**Comment by TracerDS** [comment]
> I do not know about jumping while in air do you mean jump after an object then fall in air? I think the task queue is cleared once the task is killed or no task is active consider empty

I mean, you jump then immediately after you kill the task. What would happen? Would the jump finish but without animation or would it freeze in the air for a small time and then fly straight down?

**Comment by Proxy-99** [comment]
> > I do not know about jumping while in air do you mean jump after an object then fall in air? I think the task queue is cleared once the task is killed or no task is active consider empty
> 
> I mean, you jump then immediately after you kill the task. What would happen? Would the jump finish but without animation or would it freeze in the air for a small time and then fly straight down?

you better test it than explain 😅

File: Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp (hunk 4 above)
Comment (TheNormalnij) [reply Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp]: It will be a string in Lua API, but we are using enums for such data

Comment (Proxy-99) [reply Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp]: I still do not understand it can u give me the code or something explain more

Comment (TheNormalnij) [reply Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp]: check `CLuaEngineDefs::EngineSetPoolCapacity`

Comment (Proxy-99) [reply Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp]: I am getting this error with the argument parser
```
   F:\MTA\mtasa-blue\Shared\mods\deathmatch\logic\lua\CLuaFunctionParser.h(426,54):
4>            Conversion to enumeration type requires an explicit cast (static_cast, C-style cast or parenthesized function-style cast)
4>    F:\MTA\mtasa-blue\Shared\mods\deathmatch\logic\lua\CLuaFunctionParser.h(426,35):
4>    while trying to match the argument list '(T)'
4>        with
4>        [
4>            T=eSecondaryTaskType
4>        ]
```

Comment (Proxy-99) [reply Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp]: I have fix it and add the enum here (https://github.com/multitheftauto/mtasa-blue/pull/3808/commits/65bb7bfab722d34e497fafc5b280f3cde1c81e34) @TheNormalnij

File: Client/sdk/game/CTaskManager.h [hunk 5]
```diff
@@ -41,6 +41,13 @@ enum
     ABORT_PRIORITY_IMMEDIATE
 };
 
+enum taskType
```
Comment (TheNormalnij) [inline Client/sdk/game/CTaskManager.h:44]: be consistent with another code. `eTaskType`. It's not a part of hungarian notation

Comment (Proxy-99) [reply Client/sdk/game/CTaskManager.h:44]: so what letter e reference to?

Comment (TracerDS) [reply Client/sdk/game/CTaskManager.h:44]: `e` -> `enum`

Comment (TheNormalnij) [reply Client/sdk/game/CTaskManager.h:44]: enumeration

Comment (Proxy-99) [reply Client/sdk/game/CTaskManager.h:44]: so we can use hungarian notation with enums?
I mean I don't see any benefits of using that format

Comment (Fernando-A-Rocha) [reply Client/sdk/game/CTaskManager.h:44]: > so we can use hungarian notation with enums? I mean I don't see any benefits of using that format

Using e for enumartion is not considered Hungarian notation, it's a different principle and a practiced used in all code

Comment (Proxy-99) [reply Client/sdk/game/CTaskManager.h:44]: I do not see the point of renaming this

Comment (Dutchman101) [reply Client/sdk/game/CTaskManager.h:44]: It's too trivial to block this

**Comment by Dutchman101** [comment]
Thanks, please add it to wiki
@Proxy-99

**Pull request merged**